code/game/objects/structures/lattice.dm

```diff
@@ -58,12 +58,15 @@
 			return
 		var/obj/item/tool/weldingtool/WT = C
 		if(WT.remove_fuel(0, user))
-			to_chat(user, SPAN_NOTICE(" Slicing lattice joints ..."))
-		new /obj/item/stack/rods(src.loc)
-		qdel(src)
-
+			to_chat(user, SPAN_NOTICE("Slicing lattice joints..."))
+		deconstruct()
 	return
 
+/obj/structure/lattice/deconstruct(disassembled = TRUE)
+	if(disassembled)
+		new /obj/item/stack/rods(src.loc)
+	. = ..()
```

Comment:
same thing here, do this for the rest of them in your PR since I don't want to list them all out